SeQuant/core/utility/singleton.hpp

```diff
@@ -117,12 +117,12 @@ class Singleton {
  private:
   /// Constructs a default-constructed instance. If called from multiple threads
   /// only the first call will construct the instance.
-  static std::shared_ptr<Derived> set_default_instance() {
+  static Derived& set_default_instance() {
     std::scoped_lock lock(instance_mutex());
     if (!instance_accessor()) {
-      instance_accessor() = std::move(std::shared_ptr<Derived>(new Derived));
+      instance_accessor() = std::move(std::unique_ptr<Derived>(new Derived));
```

Comment:
move should not be needed here.
Also, we probably should make use of std::make_unique

Reply:
can't use `std::make_unique` (see note above re: private ctor) ... agree that `std::move` is not needed but is there a penalty for its use here? as a visual annotation of where moves occur it beats a comment.